scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,143 @@
+        measurement_assigned = False
+        stat_var = {}
+        # part_list contains a list of tokens of the column name after splitting the
+        # column name based on the '!!' delimiter
+        part_list = column.split(self.delimiter)
+
+        # part_list[-1], the last element of part_list describes the property value.
+        # This element points to the 'v' in 'pv' -> and we map it to the
+        # corresponding property 'p'
+        # Eg:
+        # 'Estimate!!Total!!Total civilian noninstitutionalized
+        # population!!SEX!!Male' is the name of the column and the value you we want
+        # is the gender 'Male'. This pattern repeats for all columns
+        # To verify this assumption for specific census tables, please check the
+        # table shells at: https://www2.census.gov/programs-surveys/acs/tech_docs/table_shells/2019/
+        val_prop_col = part_list[-1]
+
+        for part in part_list:
+            # set the base for special cases like median, etc.
+            if not measurement_assigned and 'measurement' in self.features:
+                if part in self.features['measurement']:
+                    stat_var.update(self.features['measurement'][part])
+                    measurement_assigned = True
+
+        # set the default statVar definition
+        if not measurement_assigned and 'measurement' in self.features and '_DEFAULT' in self.features['measurement']:
+            stat_var.update(self.features['measurement']['_DEFAULT'])
+
+        # set the populationType attribute
+        if 'populationType' not in stat_var:
+          stat_var['populationType'] = self._get_population_type(part_list)
+
+        # associate pvs to stat_var
+        if 'pvs' in self.features:
+            #p = property, k = dictionary of substrings in column that points to a propertyValue
+            for p, k in self.features['pvs'].items():
+                for c, v in k.items():
+                    ## check if the current key of dict matches with any substring from the tokens of a column name
+                    for part in part_list:
+                        # check if column substring matches with a column token
+                        if c == part:
+                            stat_var[p] = v
+                            # add inferred properties from property 'p' to the stat_var
+                            if 'inferredSpec' in self.features and p in self.features['inferredSpec']:
+                                  stat_var.update(self.features['inferredSpec'][p])
+
+
+      
+        #TODO: If quantityRanges occurs as values in multiple properties, find the property that relates to the column
+        #Suggestion from Prashnath: using _DISAMBIGUATING_POP_TYPE for pvs (income and earnings -- in the test) 
+
+        #TODO: Add units and scalingFactor
+
+        # if the column is Margin of Error, update statType key
+        if 'Margin of Error' in part_list:
+            stat_var['statType'] = 'marginOfError'
```

Comment:
This seems acs-5 year census subject table specific code.
Can we move this to a separate function, say add_additional_pv() that could be overriden by other classes like other census b-tables in future to add other pvs

Reply:
Done.